sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamingQueryListenerBus.scala

```diff
@@ -41,6 +41,8 @@ class StreamingQueryListenerBus(sparkListenerBus: LiveListenerBus)
   def post(event: StreamingQueryListener.Event) {
     event match {
       case s: QueryStartedEvent =>
+        sparkListenerBus.post(s)
```

Comment:
`QueryStartedEvent` will be sent twice to `StreamingQueryListener`. There is a thread local variable in `LiveListenerBus`. You can use it to ignore duplicated  `QueryStartedEvent` if it's posted in the listener thread.